File: samples/java-rx2/src/main/java/com/example/samples/javarx2/MainActivity.java
package com.example.samples.javarx2;

import android.os.Bundle;
import android.util.Log;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.example.core.Client;
import com.example.core.response.Repo;
import com.example.javarx2.RxSingleContinuation;

import java.util.List;

import io.reactivex.Single;
import io.reactivex.disposables.Disposable;

public class MainActivity extends AppCompatActivity {

    Client client = new Client();

    Disposable disposable;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Single<? super List<? extends Repo>> single = Single.create(emitter -> {
            client.getReposAsync("benjaminledet", new RxSingleContinuation(emitter));
        });

        disposable = single.subscribe(
                result -> {
                    Log.d("MainActivity", result.toString());
                },
                error ->  {
                    Log.e("MainActivity", "getReposAsync", error);
                }
        );
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        disposable.dispose();
    }
}

File: samples/kotlin-rx2/src/main/java/com/example/samples/kotlinrx2/MainActivity.kt
package com.example.samples.kotlinrx2

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import com.example.core.Client
import io.reactivex.disposables.Disposable
import kotlinx.coroutines.rx2.rxSingle

class MainActivity : AppCompatActivity() {

    private val githubClient = Client()

    lateinit var observable: Disposable

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        observable = rxSingle { githubClient.getReposAsync("benjaminledet") }.subscribe({ result ->
            Log.d("MainActivity", result.toString())
        }, { exception ->
            Log.e("MainActivity", "getReposAsync", exception)
        })
    }

    override fun onDestroy() {
        super.onDestroy()
        observable.dispose()
    }
}
